PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -911,3 +911,26 @@ bool matchInList(const TCHAR *fileName, const std::vector<generic_string> & patt
 	return false;
 }
 
+void createFullDirectory(const generic_string &dirName)
+{
+	TCHAR tmpName[MAX_PATH];
+
+	// Create a copy so we can modify it
+	_tcscpy_s(tmpName, MAX_PATH, dirName.c_str());
+
+	// Make sure the path ends in a backslash
+	if (dirName.back() != TEXT('\\'))
+		_tcscat_s(tmpName, MAX_PATH, TEXT("\\"));
+
+	// Create parent directories
+	for (auto p = _tcschr(tmpName, TEXT('\\')); p != NULL; p = _tcschr(p + 1, TEXT('\\')))
+	{
+		*p = 0;
+		// May or may not already exist
+		::CreateDirectory(tmpName, NULL);

```

Comment:
That's really good to know! I'd much prefer to use SHCreateDirectory anyways. I'm never sure how serious to take the "warnings" from Microsoft about certain functions. I will update the PR when I get a chance.

Thanks!